File: RxCommLib/src/main/java/info/mx/comlib/retrofit/service/model/google/Bounds.java

package info.mx.comlib.retrofit.service.model.google;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

@SuppressWarnings("JavadocDeclaration")
public class Bounds implements Serializable, Parcelable {

    @SerializedName("northeast")
    @Expose
    private Northeast northeast;
    @SerializedName("southwest")
    @Expose
    private Southwest southwest;
    public final static Creator<Bounds> CREATOR = new Creator<>() {

        public Bounds createFromParcel(Parcel in) {
            return new Bounds(in);
        }

        public Bounds[] newArray(int size) {
            return (new Bounds[size]);
        }

    };
    private final static long serialVersionUID = 2058643496684590933L;

    protected Bounds(Parcel in) {
        this.northeast = ((Northeast) in.readValue((Northeast.class.getClassLoader())));
        this.southwest = ((Southwest) in.readValue((Southwest.class.getClassLoader())));
    }

    /**
     * No args constructor for use in serialization
     */
    public Bounds() {
    }

    /**
     * @param southwest
     * @param northeast
     */
    public Bounds(Northeast northeast, Southwest southwest) {
        super();
        this.northeast = northeast;
        this.southwest = southwest;
    }

    public Northeast getNortheast() {
        return northeast;
    }

    public void setNortheast(Northeast northeast) {
        this.northeast = northeast;
    }

    public Southwest getSouthwest() {
        return southwest;
    }

    public void setSouthwest(Southwest southwest) {
        this.southwest = southwest;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(northeast);
        dest.writeValue(southwest);
    }

    public int describeContents() {
        return 0;
    }

}

File: RxCommLib/src/main/java/info/mx/comlib/retrofit/service/model/google/Bounds.kt
package info.mx.comlib.retrofit.service.model.google

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.io.Serializable

open class Bounds : Serializable, Parcelable {
    @SerializedName("northeast")
    @Expose
    var northeast: Northeast? = null

    @SerializedName("southwest")
    @Expose
    var southwest: Southwest? = null

    protected constructor(`in`: Parcel) {
        this.northeast = (`in`.readValue((Northeast::class.java.classLoader)) as Northeast?)
        this.southwest = (`in`.readValue((Southwest::class.java.classLoader)) as Southwest?)
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(northeast)
        dest.writeValue(southwest)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<Bounds?> = object : Parcelable.Creator<Bounds?> {
            override fun createFromParcel(`in`: Parcel): Bounds {
                return Bounds(`in`)
            }

            override fun newArray(size: Int): Array<Bounds?> {
                return arrayOfNulls(size)
            }
        }
    }
}
